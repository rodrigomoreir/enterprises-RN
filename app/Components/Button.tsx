import React from 'react';
import { Text, TouchableOpacity, TouchableOpacityProps } from 'react-native'
import { getStyles } from './Button.style';

interface ButtonProps extends TouchableOpacityProps {
  text: string;
  onPress: () => void
}

const Button: React.FC<ButtonProps> = ({ text, onPress,  ...rest }) => { 
  const styles = getStyles()
  
  return(
    <TouchableOpacity style={styles.container} onPress={onPress} {...rest} >
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
)};

export default Button;