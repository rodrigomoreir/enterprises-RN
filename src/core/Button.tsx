import React from 'react'
import { TouchableOpacityProps, ActivityIndicator } from 'react-native'
import styled from 'styled-components/native'
import { screenWidth } from '../utils'

interface ButtonProps extends TouchableOpacityProps {
  text: string
  onPress: () => void
  isLoading: boolean
}

const Button: React.FC<ButtonProps> = ({
  text,
  onPress,
  isLoading,
  ...rest
}) => {
  return (
    <>
      {!isLoading ? (
        <StyledTouchableOpacity onPress={onPress} {...rest}>
          <StyledText>{text}</StyledText>
        </StyledTouchableOpacity>
      ) : (
        <ActivityIndicator />
      )}
    </>
  )
}

const StyledTouchableOpacity = styled.TouchableOpacity`
  width: ${screenWidth - 100}px;
  height: 60px;
  background-color: ${({ theme }) => theme.colors.blue};
  border-radius: 30px;
  justify-content: center;
  align-items: center;
`
const StyledText = styled.Text`
  font-weight: bold;
  color: ${({ theme }) => theme.colors.white};
  font-size: 18px;
`

export default Button
