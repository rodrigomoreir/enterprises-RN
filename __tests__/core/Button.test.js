import React from 'react'

// import renderer from 'react-test-renderer'
import { fireEvent } from '@testing-library/react-native'

import {
  renderWithTheme,
  matchSnapshotWithTheme,
} from '../../src/functions/testing'

import Button from '../../src/core/Button'

describe('Button', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<Button text={'TEST'} />)
  })

  test('renders correctly', () => {
    const onPressMock = jest.fn()
    const { getByText } = renderWithTheme(
      <Button onPress={onPressMock} text={'Press me'} />,
    )
    const button = getByText('Press me')
    fireEvent.press(button)
    expect(onPressMock).toHaveBeenCalledTimes(1)
    expect(button).toBeTruthy()
  })

  test('render button with loading', () => {
    renderWithTheme(<Button onPress={jest.fn()} isLoading text={'Press me'} />)
  })
})
