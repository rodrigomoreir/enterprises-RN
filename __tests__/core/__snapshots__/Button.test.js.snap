// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#87cefa",
      "borderRadius": 30,
      "height": 60,
      "justifyContent": "center",
      "opacity": 1,
      "width": 650,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#ffffff",
          "fontSize": 18,
          "fontWeight": "bold",
        },
      ]
    }
  >
    TEST
  </Text>
</View>
`;
